airflow/timetables/base.py

```diff
@@ -122,11 +122,12 @@ class Timetable(Protocol):
     like ``schedule=None`` and ``"@once"`` set it to *False*.
     """
 
-    can_run: bool = True
-    """Whether this timetable can actually schedule runs.
+    can_be_scheduled: bool = True
+    """Whether this timetable can actually schedule runs in an automated manner.
 
-    This defaults to and should generally be *True*, but ``NullTimetable`` sets
-    this to *False*.
+    This defaults to and should generally be *True* (including non periodic
+    execution types like *@once* and data triggered tables), but
+    ``NullTimetable`` sets this to *False*.
     """
```

Comment:
We’ll probably need to add some kind of compatibility layer for this, since `can_run` is technically public interface and may be implemented by an existing timetable. Something that emits a deprecation warning if `can_run` is implemented on a timetable but `can_be_scheduled` is not (and forward `can_run` to `can_be_scheduled`).

Reply:
I think the other way around is needed—since it’s Airflow that accesses the value, and the user implements the timetable, we need to detect when `can_be_scheduled` is accessed, and emit a warning when `can_run` is defined to a different value.

It’d also probably be best to not use `__getattribute__` since the function would be called on _every_ attribute access and slow down timetable access.

Reply:
I reworked the part with the deprecation warning and use now a property to return a warning if can_run is used in a timetable instead of can_be_scheduled.